### ISSUE TITLE
dockershim.sock in 1.23 only supports v1 of CRI and tools dependent on v1alpha2 will not work

### ISSUE THREAD

**Issue by SergeyKanzhelev** [kubernetes/kubernetes#106798]
### What happened?

With this change: https://github.com/kubernetes/kubernetes/pull/106501 the version of CRI API that `dockershim.sock` exposes is `v1` (as oppose to `v1alpha2` it used to be). So tools that takes dependency on dockershim.sock will fail with:

```
rpc error: code = Unimplemented desc = unknown service runtime.v1alpha2.ImageServicepanic
```

/sig node
/priority critical-urgent

### What did you expect to happen?

v1alpha2 is exposed as before. Especially given that we only have a single release left where dockershim will present

### How can we reproduce it (as minimally and precisely as possible)?

For example, this program will fail on 1.23

``` go
package main

import (
	"context"
	"fmt"

	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"

	"google.golang.org/grpc"
)

const (
    addr = "unix:///var/run/dockershim.sock"
)

func main() {
    ctx := context.Background()
    newclient, err := NewImageService()
    if err != nil {
        fmt.Printf("error with client: %v", err)
    }
    images, err := newclient.ListImages(ctx, &runtimeapi.ListImagesRequest{})
    if err != nil {
        fmt.Printf("error getting images: %v", err)
    }
    for _, img := range images.Images {
        fmt.Println(img.RepoTags)
    }

}

func NewImageService() (runtimeapi.ImageServiceClient, error) {
    conn, err := grpc.Dial(addr, grpc.WithInsecure())
    return runtimeapi.NewImageServiceClient(conn), err
}
```

### Kubernetes version

1.23

### Cloud provider

any

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by liggitt** [comment]
good catch, agree on keeping dockershim as-is in 1.23 ahead of removal

**Comment by saschagrunert** [comment]
Fix is incoming in #106803

**Comment by SergeyKanzhelev** [comment]
I closed my PR, your PR has a conversion logic which is needed
